src/main/webapp/advertisements.xhtml

```diff
@@ -40,9 +41,10 @@
                                 </p:commandButton>
                             </div>
                         </div>
-
                     </p:dataGrid>
+                </h:form>
 
+                <h:form id="adForm">
```

Comment:
Probably id could better be name "dialog-form", for better readability. Good practice is to avoid camelCase in CSS names.